addons/account/models/account_move.py

```diff
@@ -1892,6 +1899,18 @@ def _compute_taxes_legal_notes(self):
                 if not is_html_empty(tax.invoice_legal_notes)
             )
 
+    @api.depends('line_ids.payment_date', 'line_ids.reconciled')
+    def _compute_next_payment_date(self):
+        for move in self:
+            move.next_payment_date = min([line.payment_date for line in move.line_ids.filtered(lambda l: l.payment_date and not l.reconciled)], default=False)
+
+    def _search_next_payment_date(self, operator, value):
+        lines_before_date = self.env['account.move.line'].search([
+            ('reconciled', '=', False),
+            ('payment_date', operator, value)
```

Comment:
This doesn't consider only the minimum date, does it?
So this filter only makes sense with the `<` operator? Might be confusing functionally in other cases